pkg/asset/ignition/bootstrap/bootstrap_in_place.go

```diff
@@ -0,0 +1,121 @@
+	"github.com/openshift/installer/pkg/asset/manifests"
+	"github.com/openshift/installer/pkg/asset/releaseimage"
+	"github.com/openshift/installer/pkg/asset/rhcos"
+	"github.com/openshift/installer/pkg/asset/tls"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+const (
+	singleNodeBootstrapInPlaceIgnFilename = "bootstrap-in-place-for-live-iso.ign"
+)
+
+// SingleNodeBootstrapInPlace is an asset that generates the ignition config for single node OpenShift.
+type SingleNodeBootstrapInPlace struct {
+	Bootstrap *Bootstrap
+	File      *asset.File
+	Config    *igntypes.Config
+}
+
+var _ asset.Asset = (*SingleNodeBootstrapInPlace)(nil)
+
+// Dependencies returns no dependencies.
+func (a *SingleNodeBootstrapInPlace) Dependencies() []asset.Asset {
+	return a.Bootstrap.Dependencies()
+}
+
+// Name returns the human-friendly name of the asset.
+func (a *SingleNodeBootstrapInPlace) Name() string {
+	return "Single node bootstrap In Place Ignition Config"
+}
+
+// Files returns the password file.
+func (a *SingleNodeBootstrapInPlace) Files() []*asset.File {
+	if a.File != nil {
+		return []*asset.File{a.File}
+	}
+	return []*asset.File{}
+}
+
+// Generate generates the ignition config for the Bootstrap asset.
+func (a *SingleNodeBootstrapInPlace) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	proxy := &manifests.Proxy{}
+	releaseImage := &releaseimage.Image{}
+	rhcosImage := new(rhcos.Image)
+	bootstrapSSHKeyPair := &tls.BootstrapSSHKeyPair{}
+	ironicCreds := &baremetal.IronicCreds{}
+	dependencies.Get(installConfig, proxy, releaseImage, rhcosImage, bootstrapSSHKeyPair, ironicCreds)
+	if err := verifyBootstrapInPlace(installConfig.Config); err != nil {
+		return err
+	}
+	templateData, err := a.Bootstrap.getTemplateData(installConfig.Config, releaseImage.PullSpec, installConfig.Config.ImageContentSources, proxy.Config, rhcosImage, ironicCreds)
+
+	if err = a.Bootstrap.Generate(dependencies); err != nil {
+		return err
+	}
+	err = a.Bootstrap.addStorageFiles("/", "bootstrap/bootstrap-in-place/files", templateData)
+	if err != nil {
+		return err
+	}
+	err = a.Bootstrap.addSystemdUnits("bootstrap/bootstrap-in-place/systemd/units", templateData)
+	if err != nil {
+		return err
+	}
+	a.File = &asset.File{
+		Filename: singleNodeBootstrapInPlaceIgnFilename,
+		Data:     a.Bootstrap.File.Data,
+	}
+	return nil
+}
+
+// Load returns the bootstrap-in-place ignition from disk.
+func (a *SingleNodeBootstrapInPlace) Load(f asset.FileFetcher) (found bool, err error) {
+	file, err := f.FetchByName(singleNodeBootstrapInPlaceIgnFilename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	config := &igntypes.Config{}
+	if err := json.Unmarshal(file.Data, config); err != nil {
+		return false, errors.Wrapf(err, "failed to unmarshal %s", bootstrapIgnFilename)
+	}
+
+	a.File, a.Config = file, config
```

Comment:
The `Load` needs to populate `a.Bootstrap` as well, or you need to make `a.Bootstrap` private. Otherwise, the asset store will always consider the asset dirty.

The asset store has a JSON copy of the asset. When the store loads the asset from disk, the store compares the asset loaded against the JSON copy of the asset. If they are different, then the asset store considers the asset dirty and all of the assets that depend on the dirty asset to be dirty as well. In the case of the bootstrap-in-place asset, that does not have consequences since there are no other assets that depend upon it, but it should still be fixed.